File: backend/Models/Product.cs
using System;
using System.Collections.Generic;

namespace backend.Models;

public partial class Product
{
    public int ProductId { get; set; }

    public string? Name { get; set; }

    public string? Description { get; set; }

    public decimal? Price { get; set; }

    public int? Stock { get; set; }

    public DateTime? CreatedAt { get; set; }

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: frontend/src/app/shared/models/product.ts
export interface ProductResponse {
  productId: number;
  name: string | null;
  description: string | null;
  price: number | null;
  stock: number | null;
  createdAt: string | null;
}

export interface CreateProductRequest {
  name: string;
  description?: string | null;
  price?: number;
  stock?: number;
}

export interface UpdateProductRequest {
  name: string;
  description?: string | null;
  price?: number;
  stock?: number;
}
